helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -1127,7 +1066,7 @@ private void updateDisabledInstances(Collection<InstanceConfig> allInstanceConfi
     _disabledInstanceSet.clear();
     for (InstanceConfig config : allInstanceConfigs) {
       Map<String, List<String>> disabledPartitionMap = config.getDisabledPartitionsMap();
-      if (!InstanceValidationUtil.isInstanceEnabled(config, clusterConfig)) {
+      if (config.getInstanceOperation().equals(InstanceConstants.InstanceOperation.DISABLE)) {
```

Comment:
I think we should remove `InstanceValidationUtil.isInstanceEnabled` all together since that method is just calling isInstanceEnabled and skipping the logic for checking if instance is in DISABLED_INSTANCES of cluster config.

Also, controller can no longer just say `!config.isInstanceEnabled` since there are more states to consider than just ENABLE and DISABLE. We need to check if the InstanceOperation is equal to DISABLE.

Reply:
I have deprecated `InstanceValidationUtil.isInstanceEnabled` and fields for disabling instances in ClusterConfig. We will move away from that as it isn't being used currently and does not align with future plans for InstanceOperation.